openstudiocore/src/model/MeterCustom.cpp

```diff
@@ -0,0 +1,298 @@
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#include "MeterCustom.hpp"
+#include "MeterCustom_Impl.hpp"
+
+#include "Model.hpp"
+#include "Model_Impl.hpp"
+
+#include "ModelObject.hpp"
+#include "ModelObject_Impl.hpp"
+
+#include <utilities/idd/IddFactory.hxx>
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_Meter_Custom_FieldEnums.hxx>
+
+// Needed for extensible groups
+#include "../utilities/idf/WorkspaceExtensibleGroup.hpp"
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  MeterCustom_Impl::MeterCustom_Impl(const IdfObject& idfObject,
+                                     Model_Impl* model,
+                                     bool keepHandle)
+    : ModelObject_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == MeterCustom::iddObjectType());
+  }
+
+  MeterCustom_Impl::MeterCustom_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                     Model_Impl* model,
+                                     bool keepHandle)
+    : ModelObject_Impl(other, model, keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == MeterCustom::iddObjectType());
+  }
+
+  MeterCustom_Impl::MeterCustom_Impl(const MeterCustom_Impl& other,
+                                     Model_Impl* model,
+                                     bool keepHandle)
+    : ModelObject_Impl(other,model,keepHandle)
+  {}
+
+  // This logically doesn't produce any output variables
+  const std::vector<std::string>& MeterCustom_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty()){
+    }
+    LOG(Info, "Meter:Custom does not produce any output variables.");
+    return result;
+  }
+
+  IddObjectType MeterCustom_Impl::iddObjectType() const {
+    return MeterCustom::iddObjectType();
+  }
+
+  // fuel Type
+  boost::optional<std::string> MeterCustom_Impl::fuelType() const {
+    return getString(OS_Meter_CustomFields::FuelType,true);
+  }
+
+  bool MeterCustom_Impl::setFuelType(boost::optional<std::string> fuelType) {
```

Comment:
@jmarrec this should be `bool MeterCustom_Impl::setFuelType(const std::string& fuelType) {` the optional input is an error from the class generator.  This method takes a string and the reset method resets it